cloudinit/net/__init__.py

```diff
@@ -329,6 +329,47 @@ def is_disabled_cfg(cfg):
 
 def find_fallback_nic(blacklist_drivers=None):
     """Return the name of the 'fallback' network device."""
+    if util.is_FreeBSD():
+        return find_fallback_nic_on_freebsd()
+    else:
+        return find_fallback_nic_on_linux()
+
+
+def find_fallback_nic_on_freebsd(blacklist_drivers=None):
+    """Return the name of the 'fallback' network device on FreeBSD.
+
+    @param blacklist_drivers: currently ignored
+    @return default interface, or None
+
+
+    route -n show default returns something like::
+
+       route to: 0.0.0.0
+    destination: 0.0.0.0
+           mask: 0.0.0.0
+        gateway: 172.31.1.1
+            fib: 0
+      interface: vtnet0
+          flags: <UP,GATEWAY,DONE,STATIC>
+     recvpipe  sendpipe  ssthresh  rtt,msec    mtu        weight    expire
+           0         0         0         0      1500         1         0
+
+    we're interested in ``interface: vtnet0``.
+    """
+    route, stderr = util.subp(['route', '-n', 'show', 'default'])
```

Comment:
The network is supposed to be down (unless I miss something). Can we already use `route`?

Reply:
indeed.
so this is then basically a special case, if we already have some network connection

hence also https://github.com/canonical/cloud-init/pull/69

We also need to cover the case of no network.

the Linux code branch does this with a long dance to figure out which interfaces are connected, and i'm sure We can emulate at in a bsd appropriate fashion

perhaps we can even build a parser to pull all that data out into a dict and provide all the information necessary, easily accessible

Reply:
i think this is reasonable, and we should go for it

```python
ifcfg, stderr = util.subp(['ifconfig', '-a'])
# handle stderr
netcfg = {}
ifcfgiter = ifcfg.split("\n")
for l in ifcfgiter:
    m = l.search(r"^(?P<netif>\S*): flags=\d+<?P<flags>([\w+,]+,\w+)> .*")
    if m is not None:
        netif = m.group("netif")
        flags = m.group("flags")
        netcfg[netif] = { 'flags': flags.lowercase().split(",") }
        l = next(ifcfgiter) # this necessary?
        for l in ifcfgiter:
            # parse body:
            # inet, inet6
            # mac / ether address
            # group(s)
            # members of vlans or bridges
            # status
            # fill netcfg dict
```

Reply:
Or just the first line of `ifconfig -l -u ether` :-)